core/templates/pages/learner-dashboard-page/suggestion-modal/suggestion-modal-for-learner-dashboard.service.spec.ts

```diff
@@ -59,7 +59,9 @@ describe('Suggestion Modal Service For Learners Dashboard', () => {
   it('should open an ngbModal for suggestions requested' +
     ' when calling showSuggestionModal', () => {
     const modalSpy = spyOn(ngbModal, 'open').and.callFake((dlg, opt) => {
-      setTimeout(opt.beforeDismiss);
+      if (opt?.beforeDismiss !== undefined) {
+        setTimeout(opt.beforeDismiss);
+      }
```

Comment:
I don't think there is a need of `?` operator, Can you check once ?

Reply:
yes, `opt` could actually be `undefined` so there is a need for `?` operator.